src/org/ses/android/soap/BaseActivity.java

```diff
@@ -25,17 +30,47 @@ public boolean onCreateOptionsMenu(Menu menu) {
         return true;
     }
 
+    public void logOut()
+    {
+        AlertDialog.Builder builder = new AlertDialog.Builder(BaseActivity.this);
+        builder.setMessage(getString(R.string.exit_yes_no))
+                .setTitle(getString(R.string.warning))
+                .setCancelable(false)
+                .setPositiveButton(getString(R.string.answer_yes),
+                        new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialog, int id) {
+                                SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(BaseActivity.this);
+                                SharedPreferences.Editor editor = prefs.edit();
+                                editor.clear();
+                                editor.commit();
+                                Intent intent = new Intent(BaseActivity.this, PromoterLoginActivity.class);
+                                startActivity(intent);
+                                finish();
+                            }
+                        })
+                .setNegativeButton(getString(R.string.answer_no),
+                        new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialog, int id) {
+                                dialog.cancel();
+                            }
+                        });
+        AlertDialog alert = builder.create();
+        alert.show();
+    }
+
 
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         switch(item.getItemId()){
             case R.id.action_home:
-                // open up MainMenuActivity (there's probably some code that already does this)
-//                Log.v("menu", "home");
+                Intent intent = new Intent(getBaseContext(), MainMenuActivity.class);
+                startActivity(intent);
                 break;
             case R.id.action_logout:
-                // TODO get some common code that logs user out (abstract away from MainMenuActivity)
-//                Log.v("menu", "logout");
+                logOut();
```

Comment:
Nice! Is it possible to make `logOut` into a public static method and have `MainMenuActivity` call it when the log out button is pressed on that screen? This way we can avoid a little code duplication.